test/beast/core/test_buffer.hpp

```diff
@@ -593,6 +598,224 @@ test_dynamic_buffer(
         is_mutable_dynamic_buffer<DynamicBuffer_v0>{});
 }
 
+namespace subtests
+{
+template<class Generator>
+void test_conversion_v0(Generator generator)
+{
+    using DynamicBuffer_v0 = typename std::decay<decltype(generator())>::type;
+    BEAST_EXPECT(convertible_to_dynamic_buffer_v2<DynamicBuffer_v0&>::value == true);
+
+    // const references not convertible
+    BEAST_EXPECT(convertible_to_dynamic_buffer_v2<DynamicBuffer_v0 const&>::value == false);
+    BEAST_EXPECT(convertible_to_dynamic_buffer_v2<DynamicBuffer_v0 const&&>::value == false);
+
+    // cannot take ownership
+    BEAST_EXPECT(convertible_to_dynamic_buffer_v2<DynamicBuffer_v0>::value == false);
+    BEAST_EXPECT(convertible_to_dynamic_buffer_v2<DynamicBuffer_v0&&>::value == false);
+
+    auto storage = generator();
+    auto expected = std::string("Hello, World!");
+    storage.commit(
+        net::buffer_copy(
+            storage.prepare(expected.size()),
+                net::buffer(expected)));
+
+    auto buffer = dynamic_buffer(storage);
+    BEAST_EXPECTS(typeid(buffer) == typeid(dynamic_buffer_v0_proxy<DynamicBuffer_v0>),
+                  boost::typeindex::type_id_runtime(buffer).pretty_name());
+    {
+        auto result = buffers_to_string(buffer.data(0, buffer.size()));
+        BEAST_EXPECTS(result == expected, result);
+    }
+
+    // taking dynamic_buffer of the result of a dynamic buffer results in a cheap
+    // copy which refers to the same underlying data
+    auto second_derivative = dynamic_buffer(buffer);
+    BEAST_EXPECTS(typeid(second_derivative) == typeid(buffer),
+                  boost::typeindex::type_id_runtime(second_derivative).pretty_name());
+    {
+        auto result = buffers_to_string(second_derivative.data(0, second_derivative.size()));
+        BEAST_EXPECTS(result == expected, result);
+    }
+}
+
+template<class DynamicBuffer_v0>
+void test_conversion_v0()
+{
+    test_conversion_v0([]{ return DynamicBuffer_v0(); });
+}
+}
+
+inline
+std::string
+generate_reference_data(std::size_t length = 2048)
+{
+    std::string result;
+    std::random_device rng;
+    std::seed_seq sequence{rng(), rng(), rng(), rng(), rng()};
+    auto engine = std::default_random_engine(sequence);
+    auto range = std::uniform_int_distribution<unsigned int>(0, 61);
+    std::generate_n(std::back_inserter(result),
+                    length,
+                    [&]{
+                        auto code = range(engine);
+                        if (code < 10)
+                            return '0' + char(code);
+                        else if (code < 36)
+                            return 'A' + char(code - 10);
+                        else
+                            return 'a' + char(code - 36);
```

Comment:
This is overkill.. taking the time to ensure that all the output values are alphanumeric is way too try-hard!!!

Reply:
if they're not, they are difficult to compare visually.